src/test/kotlin/org/rust/ide/annotator/RsErrorAnnotatorTest.kt

```diff
@@ -4855,4 +4855,18 @@ class RsErrorAnnotatorTest : RsAnnotatorTestBase(RsErrorAnnotator::class) {
 
         fn main() {}
     """)
+
+    fun `test recursive async function E0733`() = checkErrors("""
```

Comment:
Maybe we could add more tests where the recursive call is somewhere deep inside `RsFunction` to ensure that `checkRecursiveAsyncFunction` is performing proper searching.

Reply:
Fixed a bug related to inner functions and lambdas and added tests